assembl/static2/js/app/components/navbar/navbar.jsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { Navbar } from 'react-bootstrap';
+import { compose, graphql } from 'react-apollo';
+import { I18n, Translate } from 'react-redux-i18n';
+
+import { getCurrentPhaseIdentifier, isSeveralIdentifiers, getPhaseName } from '../../utils/timeline';
+import { get } from '../../utils/routeMap';
+import { withScreenWidth } from '../common/screenDimensions';
+import { connectedUserIsAdmin } from '../../utils/permissions';
+import SectionsQuery from '../../graphql/SectionsQuery.graphql';
+import FlatNavbar from './FlatNavbar';
+import BurgerNavbar from './BurgerNavbar';
+import { APP_CONTAINER_MAX_WIDTH, APP_CONTAINER_PADDING } from '../../constants';
+import { displayModal } from '../../utils/utilityManager';
+import { getDiscussionSlug } from '../../utils/globalFunctions';
+
+const filterSection = ({ sectionType }, { hasResourcesCenter, hasSyntheses }) => {
+  switch (sectionType) {
+  case 'RESOURCES_CENTER':
+    return hasResourcesCenter;
+  case 'SYNTHESES':
+    return hasSyntheses;
+  case 'ADMINISTRATION':
+    return connectedUserIsAdmin();
+  case 'HOMEPAGE':
+    return false;
+  default:
+    return true;
+  }
+};
+
+const sectionFilter = (options = {}) => {
+  return (section) => {
+    return filterSection(section, options);
+  };
+};
+
+const sectionKey = ({ sectionType, url }) => {
+  return sectionType === 'CUSTOM' ? `${sectionType}-${url}` : sectionType;
+};
+
+const constantToCamelCase = (string) => {
```

Comment:
maybe rename to `upperToCamelCase` and add some js doc?

Reply:
Also, this could be `globalFunction` method, used in other places than just this component.

Reply:
Went for `snakeToCamel`